### PR TITLE
Issue 18392 create transformers

### PR DESCRIPTION
**Pull request by jdotcms**
This issue is basically a refactor for the BrowserAPI there the transformation to Map, has been delegated to individual Transformer for each kind of entity, folders, dotassets, files and pages

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SinglePageToMapViewTransformer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        if(page.getTitleImage().isPresent()) {
+            pageMap.put("titleImage", page.getTitleImage().get().variable());
+        }
+
+        pageMap.put("__icon__",      IconType.HTMLPAGE.iconName());
+
+        return pageMap;
+    }
+}
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SinglePageToMapViewTransformer.java:68]: test?

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleFolderToMapViewTransformer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        folderMap.put("mimeType", "");
+        folderMap.put("name", folder.getName());
+        folderMap.put("title", folder.getName());
+        folderMap.put("description", folder.getTitle());
+        folderMap.put("extension", "folder");
+        folderMap.put("hasTitleImage", StringPool.BLANK);
+        folderMap.put("__icon__", IconType.FOLDER.iconName());
+        return folderMap;
+    }
+}
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleFolderToMapViewTransformer.java:41]: test?

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleFileToMapViewTransformer.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        fileMap.put("hasTitleImage",  String.valueOf(fileAsset.getTitleImage().isPresent()));
+        if(fileAsset.getTitleImage().isPresent()) {
+            fileMap.put("titleImage", fileAsset.getTitleImage().get().variable());
+        }
+
+        fileMap.put("__icon__",       UtilHTML.getIconClass(fileAsset ));
+
+        return fileMap;
+    }
+}
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleFileToMapViewTransformer.java:91]: test?

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleDotAssetToMapViewTransformer.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        dotAssetMap.put("hasTitleImage",  String.valueOf(dotAsset.getTitleImage().isPresent()));
+        if(dotAsset.getTitleImage().isPresent()) {
+            dotAssetMap.put("titleImage", dotAsset.getTitleImage().get().variable());
+        }
+
+        dotAssetMap.put("__icon__",       UtilHTML.getIconClass(dotAsset));
+
+        return dotAssetMap;
+    }
+}
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleDotAssetToMapViewTransformer.java:84]: test?

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleDotAssetToMapViewTransformer.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        return Sneaky.sneaked(()->toMap()).get();
+    }
+
+    private Map<String, Object> toMap() throws DotSecurityException, DotDataException {
+
+        final Map<String, Object> dotAssetMap = new ContentletToMapTransformer(dotAsset).toMaps().get(0);
+        final Identifier identifier       = APILocator.getIdentifierAPI().find(dotAsset.getVersionId());
+        final String fileName = Try.of(()->dotAsset.getBinary("asset").getName()).getOrElse("unknown");
+
+        dotAssetMap.put("mimeType", APILocator.getFileAssetAPI().getMimeType(fileName));
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleDotAssetToMapViewTransformer.java:43]: We should probably stop doing this kind of this  APILocator.getHostAPI()
If we want to convert this into a testable code these should be class attributes where we can choose to inject them from the APILocator or a mock. And have  constructor marked as @VisibleForTesting

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleDotAssetToMapViewTransformer.java (hunk 4 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleDotAssetToMapViewTransformer.java:84]: It was already on the previous BrowserAPI

**Comment by wezell** [comment]
Was superseded by Fabro's changes

**Pull request closed without merging**